src/sdk/tests/query-string.test.ts

```diff
@@ -0,0 +1,32 @@
+import { describe } from '@jest/globals';
+import * as qs from '../query-string';
+
+const TEST_CASES = [
+  'name=vessel',
+  'name=vessel',
+  'list[]=one&list[]=two&name=vessel',
```

Comment:
different APIs support query strings differently, do we need to support other ways of representing lists? we might not but just making sure

e.g. outreach has `GET https://api.outreach.io/api/v2/prospects?filter[account][id]=1,2,3,5,8,13` - is the idea that the customer would treat that as 1 query param?

Reply:
yep exactly, that would become
```
[['filter[account][id]', '1,2,3,5,8,13']]
```
which we can pass around and extend. Apollo, for example, can add the `api_key` param to it 
```
`[['filter[account][id]', '1,2,3,5,8,13'], ['api_key', 'xxxxxxxxxxxxxx']`
```
without effecting the original query key/values when we join it back together:
```
?filter[account][id]=1,2,3,5,8,13&api_key=xxxxxxxxxxxxxx
```

Reply:
ok, we should make sure we document that behavior though